Documentation/tutorial/howto_create_custom_plugin.md

```diff
@@ -0,0 +1,160 @@
+    2.  The rtf files will be build as conceptual document.
+2.  Out of scope:
+    1.  No picture or other object in rtf files.
+    2.  No hyper link in rtf files.
+    3.  No metadata.
+
+Prepare
+-------
+1.  Create a new library project in `Visual Studio`.
+
+2.  Add nuget packages:  
+    * `System.Collections.Immutable` with version 1.1.37
+    * `System.Composition`with version 1.0.27
+
+3.  Add nuget package `Microsoft.DocAsCode.Plugins` same version with DocFX, or reference the project.
+
+4.  Add framework assembly reference:
+    `PresentationCore`, `PresentationFramework`, `WindowsBase`
+
+5.  Include some code files:
+    rtf to html converter: [MarkupConverter](https://github.com/mmanela/MarkupConverter)
+    sta task scheduler: [ParExtSamples](https://code.msdn.microsoft.com/ParExtSamples)
+
+Create a document processor
+---------------------------
+1.  Create a new class (RefDocumentProcessor.cs) with following code:
+    ```c#
+    [Export(typeof(IDocumentProcessor))]
+    public class RtfDocumentProcessor : IDocumentProcessor
+    {
+        // todo : implements IDocumentProcessor.
+    }
+    ```
+    
+2.  Declare we can handle `.rtf` file:
+
+    ```c#
+    public ProcessingPriority GetProcessingPriority(FileAndType file)
+    {
+        if (file.Type == DocumentType.Article && ".rtf".Equals(Path.GetExtension(file.File), StringComparison.OrdinalIgnoreCase))
+        {
+            return ProcessingPriority.Normal;
+        }
+        return ProcessingPriority.NotSupportted;
+    }
+    ```
+
+    Here we declare this processor can handle any `.rtf` file in article category with normal priority.
+    When two or more processors declare for same file, DocFX will give it to the higher priority one.
+
+3.  Load our rtf file by read all text:
+    ```c#
+    public FileModel Load(FileAndType file, ImmutableDictionary<string, object> metadata)
+    {
+        var content = new Dictionary<string, object>
+        {
+            ["conceptual"] = File.ReadAllText(Path.Combine(file.BaseDir, file.File)),
+            ["type"] = "Conceptual",
+            ["path"] = file.File,
+        };
+        return new FileModel(file, content);
+    }
+    ```
+    Since we want to threat `.rtf` file as markdown conceptual file, so we use same data structure with markdown conceptual - `Dictionary<string, object>`.
+
+4.  Implements `Save` method as following:
+    ```c#
+    public override SaveResult Save(FileModel model)
+    {
+        return new SaveResult
+        {
+            DocumentType = "Conceptual",
+            ModelFile = model.File,
+        };
+    }
+    ```
+
+5.  Keep `UpdateHref` method empty.
+
+6.  `Name` property is used to display in log, so give any constant string as you like.
+    e.g.:
+    ```c#
+    public string Name => nameof(RtfDocumentProcessor);
+    ```
+
+7.  `BuildSteps` property is next important property, we suggest implement as following:
```

Comment:
1. `next important property` sounds wierd...
2. describe where `BuildSteps` will be used in more detail?